replications/internal/queue_management.go

```diff
@@ -126,3 +132,106 @@ func (qm *durableQueueManager) UpdateMaxQueueSize(replicationID platform.ID, max
+		// Re-initialize a queue struct for each replication stream from sqlite
+		queue, err := durablequeue.NewQueue(
+			filepath.Join(qm.queuePath, r.ID.String()),
+			r.MaxQueueSizeBytes,
+			durablequeue.DefaultSegmentSize,
+			&durablequeue.SharedCount{},
+			durablequeue.MaxWritesPending,
+			func(bytes []byte) error {
+				return nil
+			},
+		)
+
+		if err != nil {
+			qm.logger.Error("failed to initialize replication stream durable queue", zap.Error(err))
+			errOccurred = true
+		}
+
+		// Open and map the queue struct to its replication ID
+		if err := queue.Open(); err != nil {
+			qm.logger.Error("failed to open replication stream durable queue", zap.Error(err), zap.String("id", r.ID.String()))
+			errOccurred = true
+		} else {
+			qm.replicationQueues[r.ID] = queue
+			qm.logger.Info("Opened replication stream", zap.String("id", r.ID.String()), zap.String("path", queue.Dir()))
+		}
+	}
+
+	if errOccurred {
+		return fmt.Errorf("startup tasks for replications durable queue management failed, see server logs for details")
+	}
+
+	// Get contents of replicationq directory
+	entries, err := os.ReadDir(qm.queuePath)
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range entries {
+		// Skip over non-relevant entries (must be a dir named with a replication ID)
+		if !entry.IsDir() {
+			continue
+		}
+
+		id, err := platform.IDFromString(entry.Name())
+		if err != nil {
+			continue
+		}
+
+		// Partial delete found, needs to be fully removed
+		if qm.replicationQueues[*id] == nil {
+			// Initialize queue struct for existing queue on disk
+			newQueue, err := durablequeue.NewQueue(
+				filepath.Join(qm.queuePath, id.String()),
+				67108860, // use default here, to allow for deletion
+				durablequeue.DefaultSegmentSize,
+				&durablequeue.SharedCount{},
+				durablequeue.MaxWritesPending,
+				func(bytes []byte) error {
+					return nil
+				},
+			)
+
+			if err != nil {
+				qm.logger.Error("failed to create durable queue struct during partial delete cleanup", zap.Error(err), zap.String("id", id.String()))
+				errOccurred = true
+			}
+
+			// Close and remove the queue
+			if err := newQueue.Close(); err != nil {
+				qm.logger.Error("failed to close durable queue during partial delete cleanup", zap.Error(err), zap.String("id", id.String()))
+				errOccurred = true
+			}
+
+			if err := newQueue.Remove(); err != nil {
+				qm.logger.Error("failed to remove durable queue during partial delete cleanup", zap.Error(err), zap.String("id", id.String()))
+				errOccurred = true
+			}
+		}
+	}
+
+	if errOccurred {
+		return fmt.Errorf("startup tasks for replications durable queue management failed, see server logs for details")
+	} else {
+		return nil
+	}
+}
+
+// CloseAll loops through all current replication stream queues and closes them without deleting on-disk resources
+func (qm *durableQueueManager) CloseAll() error {
+	for _, queue := range qm.replicationQueues {
+		if err := queue.Close(); err != nil {
+			return err
```

Comment:
Can we use the same pattern as above and try closing everything, even if something fails along the way?